nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java

```diff
@@ -290,8 +293,11 @@ public static ColdStorageContentStatus checkColdStorageContentAvailability(CoreS
                     DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
                     Instant downloadableUntil = blobStatus.getDownloadableUntil();
                     if (downloadableUntil != null) {
+                        String downloadableUntilStr = downloadableUntil.toString();
                         ctx.getProperties()
```

Comment:
No need to extract the variable here. Just use `downloadableUntil.toStrin()`. As reminder `toString` will give the date time in the ISO format `2011-12-03T10:15:30Z`. You should use this same date string format in the UI part.

Reply:
@andreacornaglia ,
I think what Salem has said is to keep `downloadableUntil.toString()` for the mail template
and `Date.from(downloadableUntil)` only for the property value

Reply:
`
.put(COLD_STORAGE_CONTENT_AVAILABLE_UNTIL_MAIL_TEMPLATE_KEY, downloadableUntil.toString());                        doc.setPropertyValue(COLD_STORAGE_CONTENT_DOWNLOADABLE_UNITL, downloadableUntilDate);                        session.saveDocument(doc);
`

for example

Reply:
indeed! thanks for the catch, I'll use your suggestion!